### ISSUE TITLE
Questions en realytime event monitor whit nsclient

### ISSUE THREAD

**Issue by sder73** [mickem/nscp#469]
Hello 
I do not wheer it is the best for my question, here or in the forum ?

I would like to know if in realtime monitoring we can do a wild card filtering
For example
message like 'Compaq Drive Array % has a new status of%'

I have also a problem with the use of maximum age and source id. I place then in different sections, and no reaction
I really need help, I am lost
Thank you:

**Comment by mickem** [comment]
Yes it can, you can find more details about the expression syntax in the documentation here: https://docs.nsclient.org/checks/#expressions

like will do substring matching and you also have regex which does regular expression matching. Please be aware about performance implications if you want to do regular expression on thousands of log record that might obviously be slow... So be sure to set scan range and order the checks in the expression or use real time version which only scans log as they happen

**Comment by mickem** [comment]
Yes it can, you can find more details about the expression syntax in the documentation here: https://docs.nsclient.org/checks/#expressions

like will do substring matching and you also have regex which does regular expression matching. Please be aware about performance implications if you want to do regular expression on thousands of log record that might obviously be slow... So be sure to set scan range and order the checks in the expression or use real time version which only scans log as they happen

**Comment by sder73** [comment]
Yes I have already have a look to this documentation
But I did find any response 
- on wildcard filtering on message
- why the options source ID and maximun age seem not working

Thank you

**Comment by mickem** [comment]
When it comes to wildcards you have these options:

![image](https://user-images.githubusercontent.com/1434697/30558649-756d27cc-9cb2-11e7-851c-0cc587cd21c8.png)

Not sure where you found "source ID"?
The keywords used for the eventlog filters are listed in the eventlog page here: https://docs.nsclient.org/reference/windows/CheckEventLog/#ok (not sure why they are not listed under filters as well, this is bug I guess)

There you have `id` for  eventlog id.

Do you have an example of maximum age and ok message not working?

// Michael Medin

**Comment by sder73** [comment]
Hello
I read "source id"again in your documentation 
Real time filter: default
![image](https://user-images.githubusercontent.com/31927199/30583989-b356c680-9d28-11e7-9b6b-b1a441747328.png)
may, I do not use the good qualifier. I want to change the source name of the server which send the nsca request.
For filtering message, I find and I use regexp thank you for the doc
and for I put in section [/settings/eventlog/real-time/filters/test-nsca] and it is working
and for ok message I confuse it with "empty message"

**Comment by mickem** [comment]
Ahh, sorry I thought you meant source id in the filter.

Here you have a minimalist working example:

```
[/modules]
CheckEventLog = enabled

[/settings/eventlog/real-time]
enabled = true


[/settings/eventlog/real-time/filters/my_alert]
log=application
destination=log
filter=level='error'
maximum age=30s
debug=true
```

Then if you run:
```
nscp test --log trace
```

And the following to inject messages:

```
eventcreate /ID 1 /L application /T ERROR /SO MYEVENTSOURCE /D "My first log"
eventcreate /ID 1 /L application /T INFORMATION /SO MYEVENTSOURCE /D "My first log"
```

You should see something akin to this:

![image](https://user-images.githubusercontent.com/1434697/31629593-693a5aa2-b2b4-11e7-87d8-e927f9e0c143.png)

* 1 Always makre sure there are not errors and that the parsed tree looks like you want it. If the filter has syntax issues nothign will work
* 2 This is how it looks when we inject an error message, it is caught and we get the `Notification 0: Application: 1 (error: My first log)` in the log.
* 3 This is the periodical "ok" message we get when there are not errors: `Notification 0: eventlog found no records`
* 4 Here we can see the output when there is a message but it does not match our filter.

**Comment by sder73** [comment]
Hello
All working fine now. Except the use of the qualifier source id. I am very interesting in it, because it arrive that the hostname is not the same as name in Centreon.
What I can do to work with this parameter ?

Thanks
